src/renderers/dom/shared/DOMProperty.js

```diff
@@ -197,33 +198,75 @@ var DOMProperty = {
   properties: {},
 
   /**
-   * Mapping from lowercase property names to the properly cased version, used
-   * to warn in the case of missing properties. Available only in __DEV__.
-   *
-   * autofocus is predefined, because adding it to the property whitelist
-   * causes unintended side effects.
-   *
-   * @type {Object}
+   * Some attributes are aliased for easier use within React. We don't
+   * allow direct use of these attributes. See DOMAttributeNames in
+   * HTMLPropertyConfig and SVGPropertyConfig.
    */
-  getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
+  aliases: {},
 
   /**
-   * All of the isCustomAttribute() functions that have been injected.
-   */
-  _isCustomAttributeFunctions: [],
-
-  /**
-   * Checks whether a property name is a custom attribute.
+   * Checks whether a property name is a writeable attribute.
    * @method
    */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+  shouldSetAttribute: function(name, value) {
+    if (DOMProperty.isReservedProp(name)) {
+      return false;
+    }
+
+    if (value === null) {
+      return true;
+    }
+
+    var lowerCased = name.toLowerCase();
+
+    // Prevent aliases, and badly cased aliases like `class` or `cLASS`
+    // from showing up in the DOM
+    if (DOMProperty.aliases.hasOwnProperty(lowerCased)) {
+      return false;
```

Comment:
I think it was to heavily steer users towards the alias. Dan wanted to avoid confusion where you pass `class`, or `stroke-dasharray`, it's technical valid and the HTML/SVG writes the way you expect, but it's not "The React Way", so you see a warning.